jsonschema.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+from collections import namedtuple
+
+Location = namedtuple(
```

Comment:
I think it would be useful to use the typed version of https://docs.python.org/3/library/typing.html#typing.NamedTuple for this.

Or, something with data marshaling built-in like:
- Pydantic: https://pydantic-docs.helpmanual.io/
- Attrs: https://www.attrs.org/en/stable/examples.html